src/languages/purebasic.js

```diff
@@ -0,0 +1,64 @@
+/*
+Language: PureBASIC
+Author: Tristano Ajmone <tajmone@gmail.com>
+Description: Syntax highlighting for PureBASIC (v.5). No inline ASM highlighting. First release (v.1.0), April 2016.
+Credits: I've taken inspiration from the PureBasic language file for GeSHi, created by Gustavo Julio Fiorenza (GuShH).
+Category: misc
+*/
+
+// Base deafult colors in PB IDE: background: #FFFFDF; foreground: #000000;
+
+function(hljs) {
+	var STRINGS = { // PB IDE color: #0080FF (Azure Radiance)
+		className: 'string',
+		begin: '(~)?"', end: '"',
+		illegal: '\\n',
+		relevance: 0
+	};
+	var CONSTANTS = { // PB IDE color: #924B72 (Cannon Pink)
+		//  "#" + a letter or underscore + letters, digits or underscores + (optional) "$"
+			className: 'symbol',
+			begin: '#[a-zA-Z_]\\w*\\$?',
+			relevance: 0
+	};
+	var PROCEDURES_RHS = { // PB IDE color: #006666 (Blue Stone)
+		// ")"
+		className: 'function',
+		begin: '\\)',
+		relevance: 0
+	};
```

Comment:
Right. Not sure I follow you here. The problem I faced was this: I couldn't get hjs to catch PB Procedures (functions) correctly — I've worked first on having hjs catch a function wholly, ie: from `xxx(` to `)`, but then I didn't manage to have it parse correctly strings, parameters, and other values passed within the braces.

I admint that I've worked on it with the intention to create (mainly) I highlighting script for a specific project oriented toward PureBASIC, trying to emulate the exact way it colors code in the language IDE — and the script works fine for that. But then I also worked to make it stick to general guidelines of being language-agnostic (in fact, the stylesheet was created to cover all possibile languages).

Mainly, the problem I couldn't solve was to have hjs reckognize a Function, excluding the braces (which I wanted to be kept in a separate colour), and then have it parse inside the braces for the expected parameters types.

How should I go on about it? I'm not quite sure if the problem has to do with "logical structure" of the final tags (ie: parameters should be a sub-element within the braces), or with the fact that splitting a function call (ie: "` xxx(`" and "`)`")as if they were 2 independent items is problematic.